packages/wallet-sdk/src/errors.test.ts

```diff
@@ -0,0 +1,205 @@
+import {
+  getErrorCode,
+  serializeError,
+  standardErrorCodes,
+  standardErrorMessage,
+  standardErrors,
+} from "./errors";
+import { JSONRPCRequest } from "./provider/JSONRPC";
+import { Web3Method } from "./relay/Web3Method";
+import { ErrorResponse, isErrorResponse } from "./relay/Web3Response";
+
+describe("errors", () => {
+  test("getErrorCode", () => {
+    expect(getErrorCode(4137)).toEqual(4137);
+
+    expect(getErrorCode({ code: 4137 })).toEqual(4137);
+    expect(getErrorCode({ errorCode: 4137 })).toEqual(4137);
+    expect(getErrorCode({ code: 4137, errorCode: 4137 })).toEqual(4137);
+
+    expect(getErrorCode({ code: "4137" })).toEqual(undefined);
+    expect(getErrorCode({ code: undefined })).toEqual(undefined);
+    expect(getErrorCode({ errorCode: "4137" })).toEqual(undefined);
+    expect(getErrorCode({ errorCode: undefined })).toEqual(undefined);
+
+    expect(getErrorCode({})).toEqual(undefined);
+    expect(getErrorCode("4137")).toEqual(undefined);
+    expect(getErrorCode(new Error("generic error"))).toEqual(undefined);
+
+    expect(getErrorCode(null)).toEqual(undefined);
+    expect(getErrorCode(undefined)).toEqual(undefined);
+
+    const errorResponse: ErrorResponse = {
+      method: Web3Method.generic,
+      errorMessage: "test error message",
+      errorCode: 4137,
+    };
+    expect(isErrorResponse(errorResponse)).toEqual(true);
+    expect(getErrorCode(errorResponse)).toEqual(4137);
+  });
+
+  test("standardErrorMessage", () => {
+    expect(standardErrorMessage(undefined)).toEqual("Unknown error");
+
+    // default error message
+    expect(
+      standardErrorMessage(standardErrorCodes.provider.userRejectedRequest),
+    ).toEqual(expect.stringContaining("rejected"));
+
+    // non-standard error code
+    expect(standardErrorMessage(0)).toEqual(
+      "Unspecified error message. This is a bug, please report it.",
+    );
+  });
+
+  test("unsupportedChain error", () => {
+    const errorWithoutChainID = standardErrors.provider.unsupportedChain();
```

Comment:
old versions of cbwallet/sdk return error as just number type, instead of Error object.
adding some extra tests to make sure it works as intended.